### PR TITLE
Support use of stem.version.Version

### LINKED ISSUE #108: Implement a get_version() method
Txtorcon can make use of a get_version() method which will call get_info('version'). If the users also have stem installed, the version will be in the format [stem.version.Version](https://stem.torproject.org/api/version.html#stem.version.Version).
The version is constructed from strings that conform to the 'new' style in the [tor version-spec](https://gitweb.torproject.org/torspec.git/tree/version-spec.txt) and so, can be compared too.

### PR DESCRIPTION
**Pull request by sammyshj**
This fixes #108

### CODE REVIEW

**Comment by sammyshj** [comment]
@meejah Please have a look at this PR. If the code looks good, I will go ahead and add tests for this

**Comment by sammyshj** [comment]
I have added tests for the support of stem.version.Version too

File: dev-requirements.txt [hunk 1]
```diff
@@ -11,3 +11,4 @@ mock
 # these are optional for users, but for testing we need them
 ipaddr
 GeoIP
+stem
```
Comment (meejah) [inline dev-requirements.txt]: At this point, perhaps we should add Tox environments that have (or do not have) the optional req's....but, see comment below.

File: txtorcon/torcontrolprotocol.py [hunk 2]
```diff
@@ -18,6 +18,11 @@
 import types
 import base64
 
+try:
+    import stem.version as stem_version
+except ImportError:
+    stem_version = None
+
 DEFAULT_VALUE = 'DEFAULT'
```
Comment (meejah) [inline txtorcon/torcontrolprotocol.py]: *perhaps* we should think about only trying the import if "use_stem=True" in TorControlProtocol. Mostly just thinking out loud here...Basically thinking longer-term about how to have stem-or-nor requirements.

**Comment by meejah** [comment]
This all looks pretty great, thanks!

I think what I'm going to do is merge all "stem stuff" into the (now horrificially outdated) "stem-integration" branch so we can play with possibly-disruptive changes to see how they feel.

So, I'll rebase stem-integration and merge this into it....

File: dev-requirements.txt (hunk 1 above)
Comment (sammyshj) [inline dev-requirements.txt]: I have added stem to tox.ini

**Comment by sammyshj** [comment]
The required changes that were pointed out by @meejah were made. Now, the `stem.version` is tried to be imported only when `use_stem` is True.
Also, the related tests have been added and all the tests pass successfully

**Comment by sammyshj** [comment]
@meejah I have added a new Tox environment `with-stem`.
Do let me know if everything looks good so that I can go ahead and squash all the commits into one.
Thanks

**Comment by meejah** [comment]
Yes, please squash it looks good so far :)

**Comment by sammyshj** [comment]
Done squashing the commits and the tests run good too :)
Will this commit be merged with `master` or as you mentioned before with the `stem-integration` branch?
Looks like the `stem-integration` branch is a whole lot of commits behind the `master` branch

**Comment by meejah** [comment]
Yeah, I tred rebasing stem-integration but it has a couple half-baked ideas so instead I'll create a new branch and merge this stuff into it. master is basically stable, and I think it's best to have a place we can try out more of the stem integration ideas together before trying to support a particular public API for it.

**Comment by meejah** [comment]
merged to ``optional-stem``
We can do integration there.
Note that I took out the class-name check; if it acts like a stem Version object, that's good enough.

**Pull request closed without merging**